setup_mws3.py

```diff
@@ -0,0 +1,168 @@
+    cur = con.cursor()
+
+    cur.execute("create database %s " % dbname)
+    cur.execute("create user %s@'localhost' identified by '%s'" % (username,password) )
+    cur.execute("grant all on %s.* to %s@'localhost'" % (dbname,username) )
+
+
+except mdb.Error, e:
+
+    print "****\nError we can not proceed:\n\n %d: %s \n****" % (e.args[0],e.args[1])
+    sys.exit(1)
+
+finally:
+
+    #return the config strings
+
+    f = open("%s/edpcmentoring/edpcmentoring/settings_local.py" % filedir_path, 'w')
+    f.write( "# This file has been generated using %s \n" % sys.argv[0] )
+    f.write( "SECRET_KEY='%s'" % secret_key )
+    config = """
+DEBUG = True
+DATABASES = {
+        'default' : {
+            'ENGINE': 'django.db.backends.mysql',
+            'NAME': '%s',
+            'USER': '%s',
+            'PASSWORD': '%s',
+            'HOST': 'localhost',   # Or an IP Address that your DB is hosted on
+            'PORT': '3306',
+        }
+}""" % (dbname, username, password)
+
+    f.write( config )
+
+    f.close()
+
+    if con:
+        con.close()
+
+
+# we should now have a settings.mws3 file
+#
+# Next create the .htaccess file and place in the docroot
+print "Generating the .htaccess file:  "
+f = open('../docroot/.htaccess','w')
+myhtaccess="""
+AddHandler wsgi-script .py
+Options FollowSymlinks ExecCGI MultiViews Indexes
+MultiviewsMatch Handlers
+RewriteEngine On
+RewriteCond %{REQUEST_FILENAME} !-f
+RewriteRule ^(.*)$ wsgi.py/$1 [QSA,PT,L]
+#Although this may set the variable - unabel to figure out how/when to retrieve it in the django project
+#SetEnv DJANGO_SECRET_KEY 'some key goe shere'
+"""
+f.write(myhtaccess);
+f.close()
+print "done\n"
+
+
+from subprocess import call
+
+# Copy the wsgi.mws3 file from the edpcmentoring directory to the docroot
+print "Copying mws3 specific wsgi.py file:   "
+src="%s/edpcmentoring/edpcmentoring/wsgi.py.mws3" % (filedir_path)
+tgt="%s/../../docroot/wsgi.py" % (filedir_path)
+print "cp %s %s " % (src,tgt)
+call(["cp", src, tgt])
+print "done\n";
+
+# install virtualenv into env dir
+print "Generating the Python environment \n"
+print "Virtual environment  "
+virtcmd="virtualenv --system-site-packages -p/usr/bin/python2.7 %s/env" % filedir_path
+call([virtcmd],shell=True)
+print "done\n"
+
+#install libraries
+print "Installing Python libraries  "
+installcmd="whoami; . %s/env/bin/activate; pip install -r %s/requirements_mws3.txt " % (filedir_path,filedir_path)
+if is_dev:
+   installcmd="%s -r %s/dev-requirements.txt" % (installcmd,filedir_path)
+subprocess_cmd(installcmd);
+print "done\n"
+
+#create the database schema
+print "Creating database tables"
+schemacmd=". %s/env/bin/activate; %s/edpcmentoring/manage.py migrate" % (filedir_path,filedir_path)
+subprocess_cmd(schemacmd)
+print "done\n"
+
+if is_dev:
+   # load the test data
+   print "Load database with test data"
+   datacmd=". %s/env/bin/activate; %s/edpcmentoring/manage.py runscript loadtestfixtures " % (filedir_path,filedir_path)
```

Comment:
Similarly, one can call management commands directly I think. This is done elsewhere in some test scripts.